woocommerce/woocommerce-ios
# [Hazmat Shipping] Map and validate hazmat data to shipping models

Partially fix #10605 

Why
==========
When a package is flagged with containing hazmat materials, we must enforce that the user selects a category before submitting the changes. Also, the Hazmat configuration should be stored and preserved between the Shipping Label Form and the Package details.

How
==========
This PR updates the internal package validation to account for the Hazmat state correctness, and also make sure that the main models containing the Package data gets updated with the Hazmat category selected.

Screen Capture
==========
https://github.com/woocommerce/woocommerce-ios/assets/5920403/68546121-a800-49ff-ad04-9ef0f6adf3bf

How to Test
==========
1. Open the order details of an order eligible for Shipping label creation
2. Start the shipping label creation flow until you reach the Package details section
3. Inside the package details view, make sure the Contains hazardous materials toggle is visible and interactable
4. Make sure the toggle expands and collapses a view that will serve as the main control options for the HAZMAT declaration
5. Verify that checking the `contains hazardous materials` without selecting a category blocks the Package changes from be submitted.
6. Select a hazmat category and hit `done`
7. Once back to the `Create Shipping Label` form, go back to the Package Details and verify that all hazmat package selection state are the same from the submission.

Update release notes:
